src/urllib3/fields.py

```diff
@@ -1,10 +1,11 @@
 from __future__ import annotations
 
 import email.utils
+import io
 import mimetypes
 import typing
 
-_TYPE_FIELD_VALUE = typing.Union[str, bytes]
+_TYPE_FIELD_VALUE = typing.Union[str, bytes, io.BytesIO, io.BufferedReader]
```

Comment:
Note to self: I want to pull this change back out of this file because it feels like quite an invasive change! And instead consolidate this in the multipart stuff for now.